frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1048,14 +1054,34 @@ pub mod pallet {
 	#[pallet::getter(fn snapshot_metadata)]
 	pub type SnapshotMetadata<T: Config> = StorageValue<_, SolutionOrSnapshotSize>;
 
-	/// Sorted (worse -> best) list of unchecked, signed solutions.
+	/// The next index to be assigned to an incoming signed submission.
+	///
+	/// We can't just use `SignedSubmissionIndices.len()`, because that's a bounded set; past its
+	/// capacity, it will simply saturate. We can't just iterate over `SignedSubmissionsMap`,
+	/// because iteration is slow. Instead, we store the value here.
+	#[pallet::storage]
+	pub(crate) type SignedSubmissionNextIndex<T: Config> = StorageValue<_, u32, ValueQuery>;
```

Comment:
Is it correct to say this and the next one should never be accessed alone and only through `SignedSubmissions` struct?

Reply:
Looking at the logic of things like `map_take`, and the fact that you have internal overlays, it is probably the case.

Reply:
Yes, that's absolutely the intent. That's also why they're as private as I can make them, and don't provide getters.

Reply:
would be nice to maybe put it in the doc IN BOLD